web/src/main/java/org/cbioportal/web/GenericAssayController.java

```diff
@@ -148,4 +150,17 @@ private void extractMolecularProfileAndSampleIds(GenericAssayDataMultipleStudyFi
             sampleIds.add(sampleMolecularIdentifier.getSampleId());
         }
     }
+
+    private List<GenericAssayData> filterEmptyGenericAssayData(List<GenericAssayData> genericAssayDataList) {
+        List<GenericAssayData> result = new ArrayList<>();
+        genericAssayDataList.forEach(g -> {
+            String value = g.getValue();
+            // filter out empty string or "NA"
+            if (StringUtils.isNotEmpty(value) && !value.equals("NA")) {
+                result.add(g);
+            }
+        });
+
+        return result;
```

Comment:
We can make this a bit cleaner

```suggestion
        return genericAssayDataList.stream()
            .filter(g -> StringUtils.isNotEmpty(g.getValue()) && !g.getValue().equals("NA"))
            .collect(Collectors.toList());
```